pythonCode.py

```diff
@@ -0,0 +1,10 @@
+import numpy
+import pandas
+from plotnine import *
+
+#read data
+data=pandas.read_csv("chickwts.txt",sep=",",header=0)
+
+#produce barplot
+barplot=ggplot(data)+theme_classic()+xlab("feed")+ylab("weight")
+barplot+geom_bar(aes(x="factor(feed)",y="weight"),stat="summary",fun_y=numpy.mean)
```

Comment:
The hypothesis testing is not complete -0.25 pts

# get the subset of data we need
sub=data.loc[data.feed.isin(['soybean','sunflower']),:]

# null model and model accounting for feed
def null(p,y):
    expected=p[0]
    sigma=p[1]
    
    nll= -1*norm(expected,sigma).logpdf(y).sum()
    return nll

def feed(p,obs):
    B0=p[0]
    B1=p[1]
    sigma=p[2]
    
    expected=B0+B1*obs.design
    
    nll=-1*norm(expected,sigma).logpdf(obs.y).sum()
    return nll

# initial parameter guesses
nullGuess=numpy.array([250,1])
feedGuess=numpy.array([250,100,1])

# fit null model
nullFit=minimize(null,nullGuess,method="Nelder-Mead",args=sub.weight)

# input for feed model with list of ones or zeroes to indicate soybean feed
input=pandas.DataFrame({'y':sub.weight,'design':0})
input.loc[sub.feed=='soybean','design']=1

# fit feed model
feedFit=minimize(feed,feedGuess,method="Nelder-Mead",args=input)

# conduct likelihood ratio test
1-chi2.cdf(x=2*(nullFit.fun-feedFit.fun),df=1)